exchanges/okx/okx.go

```diff
@@ -0,0 +1,4453 @@
+	}
+	instrumentType = strings.ToUpper(instrumentType)
+	if instrumentType == okxInstTypeSpot || instrumentType == okxInstTypeSwap ||
+		instrumentType == okxInstTypeFutures || instrumentType == okxInstTypeOption {
+		params.Set("instType", instrumentType)
+	}
+	if instrumentID != "" {
+		params.Set("instId", instrumentID)
+	}
+	if !before.IsZero() && before.Before(time.Now()) {
+		params.Set("before", strconv.FormatInt(before.UnixMilli(), 10))
+	}
+	if !after.IsZero() && after.After(time.Now()) {
+		params.Set("after", strconv.FormatInt(after.UnixMilli(), 10))
+	}
+	if limit > 0 {
+		params.Set("limit", strconv.FormatInt(limit, 10))
+	}
+	return resp, ok.SendHTTPRequest(ctx, exchange.RestSpot, getAlgoOrderHistoryEPL, http.MethodGet, common.EncodeURLValues(algoOrderHistory, params), nil, &resp, true)
+}
+
+// GetEasyConvertCurrencyList retrieve list of small convertibles and mainstream currencies. Only applicable to the crypto balance less than $10.
+func (ok *Okx) GetEasyConvertCurrencyList(ctx context.Context) (*EasyConvertDetail, error) {
+	var resp []EasyConvertDetail
+	if err := ok.SendHTTPRequest(ctx, exchange.RestSpot, getEasyConvertCurrencyListRate, http.MethodGet,
+		easyConvertCurrencyList, nil, &resp, true); err != nil {
+		return nil, err
+	}
+	if len(resp) == 1 {
+		return &resp[0], nil
+	}
+	return nil, errNoValidResponseFromServer
+}
+
+// PlaceEasyConvert onvert small currencies to mainstream currencies. Only applicable to the crypto balance less than $10.
+func (ok *Okx) PlaceEasyConvert(ctx context.Context, arg PlaceEasyConvertParam) ([]EasyConvertItem, error) {
+	if len(arg.FromCurrency) == 0 {
+		return nil, fmt.Errorf("%w, missing 'fromCcy'", errMissingRequiredParameter)
+	}
+	if arg.ToCurrency == "" {
+		return nil, fmt.Errorf("%w, missing 'toCcy'", errMissingRequiredParameter)
+	}
+	var resp []EasyConvertItem
+	return resp, ok.SendHTTPRequest(ctx, exchange.RestSpot, placeEasyConvertEPL, http.MethodPost, easyConvert, &arg, &resp, true)
+}
+
+// GetEasyConvertHistory retrives the history and status of easy convert trades.
+func (ok *Okx) GetEasyConvertHistory(ctx context.Context, after, before time.Time, limit int64) ([]EasyConvertItem, error) {
+	params := url.Values{}
+	if !before.IsZero() {
+		params.Set("before", strconv.FormatInt(before.Unix(), 10))
+	}
+	if !after.IsZero() {
+		params.Set("after", strconv.FormatInt(after.Unix(), 10))
+	}
+	if limit > 0 {
+		params.Set("limit", strconv.FormatInt(limit, 10))
+	}
+	var resp []EasyConvertItem
+	return resp, ok.SendHTTPRequest(ctx, exchange.RestSpot, getEasyConvertHistoryEPL, http.MethodGet, easyConvertHistoryPath, nil, &resp, true)
+}
+
+// GetOneClickRepayCurrencyList retrives list of debt currency data and repay currencies. Debt currencies include both cross and isolated debts.
+// debt level "cross", and "isolated" are allowed
+func (ok *Okx) GetOneClickRepayCurrencyList(ctx context.Context, debtType string) ([]CurrencyOneClickRepay, error) {
```

Comment:
Please add coverage

Reply:
This endpoint has an issue they have to resolve. It is requiring a parameter value that doesn't exist in the documentation. Even with the parameter provided, It keeps returning the same error message.

Reply:
While this is in contrast to what I said about other tests yesterday, to me I think it would be best to still test these endpoints but expect the acctLv error. Same goes for the other One Click functions

My interpretation of this error is that you can only perform these actions at a certain account level, mine being "2" still doesn't allow it. From here: https://www.okx.com/docs/broker_en/#fd-api-and-oauth-broker it may even be a broker style feature.